generate/types.go

```diff
@@ -22,6 +22,17 @@ type goType interface {
 	// Reference returns the Go name of this type, e.g. []*MyStruct, and may be
 	// used to refer to it in Go code.
 	Reference() string
+
+	// Remove slice/pointer wrappers, and return the underlying (named (or
+	// builtin)) type.  For example, given []*MyStruct, return MyStruct.
+	Unwrap() goType
+
+	// Count the number of times Unwrap() will unwrap a slice type.  For
+	// example, given []*[]**[]MyStruct, return 3.
```

Comment:
nit: this type isn't supported in the current design; it seems like it would be nice to use a valid example (even though the method could in principle support this type as well).